Makefile

```diff
@@ -30,40 +31,45 @@ CUTVALUEFLAG_FLAGS = -DCUT_VALUE_DEBUG
 PRIMITIVEFLAG = 
 PRIMITIVEFLAG_FLAGS = -DPRIMITIVE_STUDY
 
-all: $(ROOUTIL) efficiency $(EXES)
+all: rooutil efficiency $(EXES)
 
 
 cutvalue: CUTVALUEFLAG = ${CUTVALUEFLAG_FLAGS}
-cutvalue: $(ROOUTIL) efficiency $(EXES)
+cutvalue: rooutil efficiency $(EXES)
 
 primitive: PRIMITIVEFLAG = ${PRIMITIVEFLAG_FLAGS}
-primitive: $(ROOUTIL) efficiency $(EXES)
+primitive: rooutil efficiency $(EXES)
 
 cutvalue_primitive: CUTVALUEFLAG = ${CUTVALUEFLAG_FLAGS}
 cutvalue_primitive: PRIMITIVEFLAG = ${PRIMITIVEFLAG_FLAGS}
-cutvalue_primitive: $(ROOUTIL) efficiency $(EXES)
-
+cutvalue_primitive: rooutil efficiency $(EXES)
 
 bin/doAnalysis: bin/doAnalysis.o $(OBJECTS)
-	$(CXX) $(PTCUTFLAG) $(LDFLAGS) $^ $(ROOTLIBS) $(EXTRACFLAGS) $(CUTVALUEFLAG) $(PRIMITIVEFLAG) $(EXTRAFLAGS) $(DOQUINTUPLET) $(ALPAKAINCLUDE) $(ALPAKASERIAL) -o $@
+	$(CXX) $(PTCUTFLAG) $(LDFLAGS) $^ $(ROOTLIBS) $(EXTRACFLAGS) $(CUTVALUEFLAG) $(PRIMITIVEFLAG) $(EXTRAFLAGS) $(DOQUINTUPLET) $(ALPAKAINCLUDE) $(ALPAKA_CPU) -o $@
 
-bin/sdl: bin/sdl.o $(OBJECTS)
-	$(CXX) $(PTCUTFLAG) $(LDFLAGS) $^ $(ROOTLIBS) $(EXTRACFLAGS) $(CUTVALUEFLAG) $(PRIMITIVEFLAG) $(EXTRAFLAGS) $(DOQUINTUPLET) $(ALPAKAINCLUDE) $(ALPAKASERIAL) -o $@
+bin/sdl_cpu: SDLLIB=-lsdl_cpu
+bin/sdl_cpu: bin/sdl_cpu.o $(OBJECTS_CPU)
+	$(CXX) $(PTCUTFLAG) $(LDFLAGS) $^ $(ROOTLIBS) $(EXTRACFLAGS) $(CUTVALUEFLAG) $(PRIMITIVEFLAG) $(EXTRAFLAGS) $(DOQUINTUPLET) $(ALPAKAINCLUDE) $(ALPAKA_CPU) -o $@
+bin/sdl_cuda: SDLLIB=-lsdl_cuda
+bin/sdl_cuda: bin/sdl_cuda.o $(OBJECTS_CUDA)
+	$(CXX) $(PTCUTFLAG) $(LDFLAGS) $^ $(ROOTLIBS) $(EXTRACFLAGS) $(CUTVALUEFLAG) $(PRIMITIVEFLAG) $(EXTRAFLAGS) $(DOQUINTUPLET) $(ALPAKAINCLUDE) $(ALPAKA_CUDA) -o $@
 
-%.o: %.cc
-	$(CXX) $(PTCUTFLAG) $(CFLAGS) $(EXTRACFLAGS) $(CUTVALUEFLAG) $(PRIMITIVEFLAG) $(DOQUINTUPLET) $(ALPAKAINCLUDE) $(ALPAKASERIAL) $< -c -o $@
+%_cpu.o: %.cc rooutil
+	$(CXX) $(PTCUTFLAG) $(CFLAGS) $(EXTRACFLAGS) $(CUTVALUEFLAG) $(PRIMITIVEFLAG) $(DOQUINTUPLET) $(ALPAKAINCLUDE) $(ALPAKA_CPU) $< -c -o $@
+%_cuda.o: %.cc rooutil
+	$(CXX) $(PTCUTFLAG) $(CFLAGS) $(EXTRACFLAGS) $(CUTVALUEFLAG) $(PRIMITIVEFLAG) $(DOQUINTUPLET) $(ALPAKAINCLUDE) $(ALPAKA_CUDA) $< -c -o $@
 
-$(ROOUTIL):
+rooutil:
 	$(MAKE) -C code/rooutil/
 
-efficiency:
+efficiency: rooutil
 	$(MAKE) -C efficiency/
 
 clean:
-	rm -f $(OBJECTS) bin/*.o $(EXES)
+	rm -f $(OBJECTS) bin/*.o $(EXES) bin/sdl
```

Comment:
`bin/sdl` is a symlink that gets switched to point to the binary with the selected backend. I think Slava just added it so that the CI works without any changes, but I'd be fine with removing it and adjusting the CI.

Reply:
I see now, I missed the lines in `sdl_make_tracklooper`. No changes required, I just wanted to understand how it worked. I am resolving the two related comments.